server/routes/termdb.termbyid.ts

```diff
@@ -0,0 +1,64 @@
+// import { gettermbyidRequest, gettermbyidResponse } from '#shared/types/routes/termdb.termbyid'
+import { copy_term } from '#src/termdb.js'
+
+export const api: any = {
+	endpoint: 'termdb/termbyid',
+	methods: {
+		get: {
+			init,
+			request: {
+				typeId: 'gettermbyidRequest'
+			},
+			response: {
+				typeId: 'gettermbyidResponse'
+			},
+			examples: [
+				{
+					request: {
+						body: {
+							genome: 'hg38-test',
+							dslabel: 'TermdbTest',
+							embedder: 'localhost',
+							gettermbyid: 'subcohort'
+						}
+					},
+					response: {
+						header: { status: 200 }
+					}
+				}
+			]
+		},
+		post: {
+			alternativeFor: 'get',
+			init
+		}
+	}
+}
+
+function init({ genomes }) {
+	return async (req: any, res: any): Promise<void> => {
+		const q = req.query // as getcategoriesRequest
+		try {
+			const g = genomes[req.query.genome]
+			const ds = g.datasets[req.query.dslabel]
+			const tdb = ds.cohort.termdb
```

Comment:
@karishma-gangwani just to point out this logical err. if ds is undefined, this line crashes server

Reply:
@xzhou82  I can fix this in my branch

Reply:
okay. thanks for pointing it out. were you able to fix it directly in master?